### DISCUSSION TITLE
Is there a way for the Telegram bot to read the messages first and not allow them in the chat?

### DISCUSSION THREAD

**Discussion by meecosha** [aiogram/aiogram#1296] [Q&A]
### Discussed in https://github.com/aiogram/aiogram/discussions/1294

<div type='discussions-op-text'>

<sup>Originally posted by **meecosha** September  6, 2023</sup>
What I expect:

User sends a message to a group
Bot sees it first and deletes it
Nobody else in the group is aware that there was a message sent

Я бы и на русском продублировал, но думаю, всё и так понятно</div>

**Comment by JrooTJunior** [comment]
Just handle message in the chat and delete it.
You can check that this message is first in the chat sent by user using database and handling join event (ChatMemberUpdated with enter transition)
